internal/services/api/post_test_credentials.go

```diff
@@ -37,26 +39,57 @@ func PostTestCredentialsHandlerFunc(log logging.Logger) http.HandlerFunc {
 			BadRequest(w, req, log, err)
 			return
 		}
-		acct := &accounts.Account{
-			ServerType: accounts.ServerTypeConnect,
-			AuthType:   accounts.AuthTypeAPIKey,
-			URL:        b.URL,
-			Insecure:   b.Insecure,
-			ApiKey:     b.ApiKey,
-		}
 
-		timeout := time.Duration(max(b.Timeout, 30) * 1e9)
-		client, err := clientFactory(acct, timeout, nil, log)
+		// create a list of URLs to attempt
+		possibleURLs, err := util.GetListOfPossibleURLs(b.URL)
 		if err != nil {
-			InternalError(w, req, log, err)
+			BadRequest(w, req, log, err)
 			return
 		}
 
-		user, err := client.TestAuthentication(log)
+		// walk the possible URL list backwards (it has the full path first)
+		var urlToBeTested string
+		var lastTestError error
+		var user *connect.User
+
+		for i := len(possibleURLs) - 1; i >= 0; i-- {
+			urlToBeTested = possibleURLs[i]
+
+			acct := &accounts.Account{
+				ServerType: accounts.ServerTypeConnect,
+				AuthType:   accounts.AuthTypeAPIKey,
+				URL:        urlToBeTested,
+				Insecure:   b.Insecure,
+				ApiKey:     b.ApiKey,
+			}
+
+			timeout := time.Duration(max(b.Timeout, 30) * 1e9)
+			client, err := clientFactory(acct, timeout, nil, log)
+			if err != nil {
+				InternalError(w, req, log, err)
+				return
+			}
+
+			user, lastTestError = client.TestAuthentication(log)
+			if lastTestError == nil {
+				// If we succeeded, pass back what URL succeeded
+				response := &PostTestCredentialsResponseBody{
+					User:  user,
+					Error: nil,
+					URL:   urlToBeTested,
+				}
+				w.Header().Set("content-type", "application/json")
+				w.WriteHeader(http.StatusOK)
+				json.NewEncoder(w).Encode(response)
```

Comment:
It should be possible to use `JsonResult` here

Reply:
Sorry, but I am not understanding why we would want to do that. This is a normal 200 response for this API call and my updates simply added another data field to the response.